mmv1/third_party/terraform/resources/resource_compute_instance.go

```diff
@@ -641,6 +641,29 @@ func resourceComputeInstance() *schema.Resource {
 					},
 				},
 			},
+			"advanced_machine_features": {
+				Type:        schema.TypeList,
+				MaxItems:    1,
+				Optional:    true,
+				Computed:    true,
+				Description: `Controls for advanced machine-related behavior features.`,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"enable_nested_virtualization": {
+							Type:        schema.TypeBool,
+							Optional:    true,
+							Default:     false,
+							Description: `Whether to enable nested virtualization or not.`,
+						},
+						"threads_per_core": {
+							Type:        schema.TypeInt,
+							Optional:    true,
+							Computed:    false,
+							Description: `The number of threads per physical core. To disable simultaneous multithreading (SMT) set this to 1. If unset, the maximum number of threads supported per core by the underlying processor is assumed.`,
```

Comment:
This description could probably be clearer. 

Perhaps: 
> The number of threads per physical core. To disable simultaneous multithreading (SMT) set this to 1. If unset, the maximum number of threads per core supported by the underlying processor is used by default. The number of threads per core is not configurable on machine types that have fewer than 2 vCPUs.`,

It's unclear to me whether this field will be populated with the maximum number of threads per core supported by the underlying processor or not. I assume, since it's marked as not computed that the API never returns a value that wasn't specified by the user. Is this correct?

It's also unclear from the docs when exactly the API will error. Does it allow specifying `1` for a 1 vCPU machine type, like n1-standard-1, or is it actually unconfigurable and error if any value is specified in that case? We might need to change "not configurable" to "must be `1` or unspecified".